src/cpp/session/modules/SessionPythonEnvironments.R

```diff
@@ -101,6 +101,27 @@
    if (file.exists(prefsPython))
       return(path.expand(prefsPython))
    
+   # check for Python on the PATH -- if we have an appropriate version of
+   # python3 or python on the PATH, then use it. we ignore python installations
+   # in /usr/bin because we primarily want to target python installations that
+   # appear to have been "intentionally" installed by the user. in addition,
+   # many Linux package managers install "minimized" versions of Python onto
+   # the system path that will fail to work without extra supporting packages
+   # installed, and managing and communicating all that state to the user is
+   # challenging
+   python3 <- Sys.which("python3")
```

Comment:
It feels to me like on Linux we may want to allow `/usr/bin/python3` to work as many linux users will just `apt-get` install the additional requirements they need (rather than get a version from PSF (which is only available as a source tarball: https://www.python.org/downloads/source/). In other words, I think if we do this then we effectively become "conda only" on Linux and I don't think that's what we want to do.

Reply:
Do you think we should do any pre-flight checks (e.g. checking that `pip` and `venv` are available / functional)?

Reply:
I also wonder if we should still consider preferring Anaconda in that situation; e.g. if the first copy of Python on the path is in `/usr/bin`, but Anaconda is installed, we should use Anaconda. That is, the order would be:

- Python on the PATH from a custom location (not /usr/bin),
- Anaconda Python if found,
- Python in /usr/bin

Reply:
Yes, I agree with this order for sure. I also agree that we should look for pip as a pre-flight check (not as sure about venv, let's discuss)

Reply:
As discussed, no pre-flight check necessary.